### PR TITLE
Fix url parsing for records using host placeholders

### PR DESCRIPTION
**Pull request by spape**
*PATCH VERSION*

Problem:

```ruby
class Entry < LHS::Record
  endpoint ':datastore/entry/v1/:id.json'
end
``` 

Records configured with host placeholders (see above) where raising URL.parsing exceptions, as they are per se not real urls by any standards.

Solution:

Computing urls from parameters when placeholders are detected, before the urls are actually parsed.

### CODE REVIEW

File: spec/record/includes_all_spec.rb [hunk 1]
```diff
@@ -211,5 +211,54 @@ class Contract < LHS::Record
...
+      let!(:entry_request) do
+        stub_request(:get, %r{http://datastore/entry/v1/\d+.json})
+          .to_return(
+            body: {
+              name: 'Casa Ferlin'
+            }.to_json
+          )
+      end
+
+      it 'loads included identifyable records withou raising exceptions' do
```
Comment (crackofdusk) [inline spec/record/includes_all_spec.rb]: Typos:
identif_i_able
 withou_t_

File: spec/record/includes_all_spec.rb [hunk 2]
```diff
@@ -211,5 +211,54 @@ class Contract < LHS::Record
         end
       end
     end
+
+    context 'include a known/identifyable record' do
```
Comment (crackofdusk) [inline spec/record/includes_all_spec.rb]: Typo: identif_i_able

File: spec/record/includes_all_spec.rb [hunk 3]
```diff
@@ -211,5 +211,54 @@ class Contract < LHS::Record
...
+            }.to_json
+          )
+      end
+
+      let!(:contracts_request) do
+        stub_request(:get, %r{http://datastore/contracts/\d+})
+          .to_return(
+            body: {
+              type: 'contract',
+              entry: { href: 'http://datastore/entry/v1/1.json' }
```
Comment (ryabrody) [inline spec/record/includes_all_spec.rb]: Just a general question: When does a user need to add `json` prefix?

Comment (spape) [inline spec/record/includes_all_spec.rb]: When the endpoint does not support correct acceptance headers for example and only relies on format in the url (which is not recommended anyway)

File: spec/record/includes_all_spec.rb (hunk 2 above)
Comment (spape) [inline spec/record/includes_all_spec.rb]: incred_i_ble 
thanks ;)

**Comment by spape** [comment]
Fixed the issues in SB too ;)

**Pull request merged**